pandas/core/ops/array_ops.py

```diff
@@ -150,6 +152,8 @@ def na_arithmetic_op(left, right, op, str_rep: str):
     try:
         result = expressions.evaluate(op, str_rep, left, right)
     except TypeError:
+        if is_cmp:
```

Comment:
what hits this AND is a is_cmp? can you add a comment

Reply:
so we have 7 cases that get here.  4 of these are something like `ndarray[float] > datetime`, and would pass if we removed this is_cmp check.  Of the remaining three:

- 2 involves passing a DatetimeArray to masked_arith_op, which expects ndarray.  this should probably be prevented.
- one involves comparions of complex numbers, which numpy does differently than python (which is its own PITA).  falling through to the masked op would break test_bool_flex_frame_complex_dtype.